src/components/pages/Onboarding/InviteVideoPreview.tsx

```diff
@@ -115,17 +130,26 @@ class InviteVideoPreviewView extends React.Component<
           errorMessage: "Could not upload. Please try again.",
           uploadStatus: UploadStatus.NOT_STARTED
         }),
-      () => {
+      async () => {
         this.setState({ uploadStatus: UploadStatus.UPLOADED });
-        this.requestInvite();
+        const videoDownloadUrl = await uploadTask.snapshot.ref.getDownloadURL();
+        if (videoDownloadUrl) {
+          this.sendInviteRequest(videoDownloadUrl);
```

Comment:
Should we wait to set it to `UPLOADED` here and do `UPLOADING` before?

Reply:
Yeah!